public/bank.js

```diff
@@ -82,21 +82,49 @@ $(function() {
 		},
 	});
 
+	function build_autocomplete_item(instance, item) {
+		var variable_name = instance + ':' + item.name;
+		return { key: variable_name + ')', value: variable_name, label: item.label };
+	}
+
+	// Listen on initial variable definitions broadcast
 	socket.on('variable_instance_definitions_get:result', function (err, data) {
 		if (data) {
 			var auto_complete_list = [];
 			for (var instance in data) {
 				for (var index in data[instance]) {
-					var item = data[instance][index];
-					var variable_name = instance + ':' + item.name;
-					auto_complete_list.push({ key: variable_name + ')', value: variable_name, label: item.label });
+					var item = build_autocomplete_item(instance, data[instance][index]);
+					auto_complete_list.push(item);
 				}
 			}
 
 			variables_autocomplete.append(0, auto_complete_list, true);
 		}
 	});
 
+	// Listen on instance enable/disable events
+	socket.on('variable_instance_definitions_set', function (instance, data) {
+		if (instance && data) {
+			var auto_complete_list = variables_autocomplete.collection[0].values;
+
+			// remove existing variables of instance
+			var index = 0;
+			while (index < auto_complete_list.length) {
```

Comment:
Weird way of writing for(var index = 0; index < autocoplete_list.length; ++i) 😝

Reply:
I am not sure how a list in javascript behaves, when deleting items while iterating over it. That's my solution to be on the safe side.

Reply:
the for loop I used as an example does exactly the same as your version. Remember that a `for` instruction is just 3 expressions.

So I would have done:
```
for (var i = 0; i < auto_complete_list.length; ++i) {
  if (auto_complete_list[i].value.startsWith(instance + ':')) {
    auto_complete_list.splice(i--, 1);
  }
}
```

In regards to how js behaves, the while() and the for() will both read auto_complete_list.length for each iteration, so it is just up to you to keep the index variable correct. So we both make sure that the variable is not increased when you delete :)

Just a tip!